### ISSUE TITLE
[RPC] Should `POCL_DEVICES` by case-insenstive?

### ISSUE THREAD

**Issue by Oblomov** [pocl/pocl#1078]
POCL_DEVICES is currently case-sensitive, so e.g. if you want to request a Vulkan device you have to write  `POCL_DEVICES=VULKAN` (ditto for CUDA). From a user perspective, I would expect the lowercase string to work as well. Is there a specific reason for using `strcmp`  rather than `strcasecmp` for the comparison? If not, I would gladly provide a patch to change this.

**Comment by pjaaskel** [comment]
Yeah, I just noticed the same thing while testing your PR. For now, I changed all of them to lower case for consistency (and to match what's written in the user manual), but I don't see why not it could be case insensitive as well.
